CMakeLists.txt

```diff
@@ -778,45 +778,82 @@ function(add_library_variants)
+    armv5te         ""                  "-march=armv5te"                                        "target=armv5e-none-unknown-eabi mfpu=none"                                                                "-M musicpal -cpu arm926"
+    armv6m          soft_nofp           "-mfloat-abi=soft -march=armv6m"                        "target=thumbv6m-none-unknown-eabi mfpu=none"                                                              "-M microbit"
+    armv7m          soft_nofp           "-mfloat-abi=soft -march=armv7m+nofp"                   "target=thumbv7m-none-unknown-eabi mfpu=none"                                                              "-M mps2-an385 -cpu cortex-m3"
+    armv7em         soft_nofp           "-mfloat-abi=soft -march=armv7em -mfpu=none"            "target=thumbv7em-none-unknown-eabi mfpu=none"                                                             "-M mps2-an386 -cpu cortex-m4"
+    armv7em         hard_fpv4_sp_d16    "-mfloat-abi=hard -march=armv7em -mfpu=fpv4-sp-d16"     "target=thumbv7em-none-unknown-eabihf mfpu=fpv4-sp-d16 mfloat-abi=hard"                                    "-M mps2-an386 -cpu cortex-m4"
+    armv7em         hard_fpv5_d16       "-mfloat-abi=hard -march=armv7em -mfpu=fpv5-d16"        "target=thumbv7em-none-unknown-eabihf mfpu=fpv5-d16 mfloat-abi=hard"                                       "-M mps2-an500 -cpu cortex-m7"
+    armv8m.main     soft_nofp           "-mfloat-abi=soft -march=armv8m.main+nofp"              "target=thumbv8m.main-none-unknown-eabi mfpu=none"                                                         "-M mps2-an505 -cpu cortex-m33"
+    armv8m.main     hard_fp             "-mfloat-abi=hard -march=armv8m.main+fp"                "target=thumbv8m.main-none-unknown-eabihf mfpu=fpv5-d16 mfloat-abi=hard"                                   "-M mps3-an524 -cpu cortex-m33"
+    armv8.1m.main   soft_nofp_nomve     "-mfloat-abi=soft -march=armv8.1m.main+nofp+nomve"      "target=thumbv8.1m.main-none-unknown-eabi mfpu=none"                                                       "-M mps3-an547 -cpu cortex-m55"
+    armv8.1m.main   hard_fp             "-mfloat-abi=hard -march=armv8.1m.main+fp"              "target=thumbv8.1m.main-none-unknown-eabihf march=+fp16 mfpu=fp-armv8-fullfp16-sp-d16 mfloat-abi=hard"     "-M mps3-an547 -cpu cortex-m55"
+    armv8.1m.main   hard_nofp_mve       "-mfloat-abi=hard -march=armv8.1m.main+nofp+mve"        "target=thumbv8.1m.main-none-unknown-eabihf march=+dsp march=+mve mfpu=none mfloat-abi=hard"               "-M mps3-an547 -cpu cortex-m55"
+)
+
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/multilib.yaml.in
+    ${CMAKE_CURRENT_BINARY_DIR}/multilib.yaml
+    @ONLY
+)
+install(
+    FILES ${CMAKE_CURRENT_BINARY_DIR}/multilib.yaml
+    DESTINATION lib/clang-runtimes
+    COMPONENT llvm-toolchain
+)
+
+add_custom_target(multilib_yaml DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/multilib.yaml)
+add_dependencies(llvm-toolchain multilib_yaml)
```

Comment:
It's not clear to me that this would have any effect. If it does, please add a comment explaining what it is.